inc/lift/SharedRequest.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <memory>
+
+namespace lift {
+
+class RequestPool;
+class Request;
+
+class SharedRequest
+{
+public:
+    SharedRequest(RequestPool& request_pool, std::unique_ptr<Request> request_handle);
+    SharedRequest(const SharedRequest&) = delete;
+    SharedRequest(SharedRequest&&) = delete;
+    auto operator=(const SharedRequest&) = delete;
+    auto operator=(SharedRequest&&) = delete;
+    ~SharedRequest();
+    
+    /**
+     * @return Access to the underlying request handle.
+     * @{
+     */
+    [[nodiscard]]
+    auto GetAsReference() -> Request& { return *m_request; }
+    [[nodiscard]]
+    auto GetAsReference() const -> const Request& { return *m_request; }
+    [[nodiscard]]
+    auto GetAsPointer() -> Request* { return m_request.get(); }
+    [[nodiscard]]
+    auto GetAsPointer() const -> const Request* { return m_request.get(); }
```

Comment:
Why not just `Get()` returning a pointer and the caller can deref if desired?

Reply:
That would work too, but I'm not sure what the benefit is over this.

Reply:
Less permutations of the same function :)